Gems/Atom/Tools/MaterialCanvas/Code/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+if(NOT PAL_TRAIT_BUILD_HOST_TOOLS)
+    return()
+endif()
+
+o3de_pal_dir(pal_source_dir ${CMAKE_CURRENT_LIST_DIR}/Source/Platform/${PAL_PLATFORM_NAME} ${gem_restricted_path} ${gem_path} ${gem_parent_relative_path})
+
+include(${pal_source_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake) # PAL_TRAIT_ATOM_MATERIAL_CANVAS_APPLICATION_SUPPORTED
+
+if(NOT PAL_TRAIT_ATOM_MATERIAL_CANVAS_APPLICATION_SUPPORTED)
+    return()
+endif()
+
+ly_add_target(
+    NAME MaterialCanvas EXECUTABLE
+    NAMESPACE Gem
+    AUTOMOC
+    AUTORCC
+    FILES_CMAKE
+        materialcanvas_files.cmake
+        ${pal_source_dir}/platform_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+    PLATFORM_INCLUDE_FILES
+        ${pal_source_dir}/tool_dependencies_${PAL_PLATFORM_NAME_LOWERCASE}.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            .
+            Source
+            ${pal_source_dir}
+        PUBLIC
+            Include
+    BUILD_DEPENDENCIES
+        PRIVATE
+            Gem::AtomToolsFramework.Static
+            Gem::AtomToolsFramework.Editor
+            Gem::Atom_RHI.Public
+            Gem::Atom_RHI.Reflect
+            Gem::Atom_RPI.Edit
+            Gem::Atom_RPI.Public
+            Gem::Atom_Feature_Common.Public
+            Gem::Atom_Component_DebugCamera.Static
+            Gem::AtomLyIntegration_CommonFeatures.Static
+            Gem::GraphCanvasWidgets
+            Gem::GraphModel.Editor.Static
+    RUNTIME_DEPENDENCIES
+        Gem::AtomToolsFramework.Editor
+        Gem::EditorPythonBindings.Editor
+        Gem::GraphCanvasWidgets
+        Gem::GraphCanvas.Editor
+        Gem::GraphModel.Editor
+)
+
+ly_set_gem_variant_to_load(TARGETS MaterialCanvas VARIANTS Tools)
+
+# Add a 'builders' alias to allow the MaterialCanvas root gem path to be added to the generated
+# cmake_dependencies.<project>.assetprocessor.setreg to allow the asset scan folder for it to be added
+ly_create_alias(NAME MaterialCanvas.Builders NAMESPACE Gem)
+
+# Add build dependency to Editor for the MaterialCanvas application since
+# Editor opens up the MaterialCanvas
+ly_add_dependencies(Editor Gem::MaterialCanvas)
+
+# Inject the project path into the MaterialCanvas VS debugger command arguments if the build system being invoked
+# in a project centric view
+if(NOT PROJECT_NAME STREQUAL "O3DE")
+    set_property(TARGET MaterialCanvas APPEND PROPERTY VS_DEBUGGER_COMMAND_ARGUMENTS "--project-path=\"${CMAKE_SOURCE_DIR}\"")
+endif()
+
+# Adds the MaterialCanvas target as a C preprocessor define so that it can be used as a Settings Registry
+# specialization in order to look up the generated .setreg which contains the dependencies
+# specified for the target.
+if(TARGET MaterialCanvas)
+    set_source_files_properties(
+        Source/MaterialCanvasApplication.cpp
+        PROPERTIES
+            COMPILE_DEFINITIONS
+                LY_CMAKE_TARGET="MaterialCanvas"
```

Comment:
I suggest we label this "MaterialCanvasPrototype" or something like that until it's in a usable state.

Reply:
Keeping the project and target name but I made the window title very explicit with your other suggestion